backend/src/services/block.service.ts

```diff
@@ -55,17 +59,36 @@ export class BlockService {
     };
   }
 
+  public async getLastMinedBlocks(): Promise<BaseBlockResponse> {
+    const allBlocks = await this.blockStorage.getAllBlocks();
+    const latestMinedBlock =
+    allBlocks && allBlocks.length
+      ? {
+          hash: allBlocks[allBlocks.length - 1].hash,
+          number: allBlocks[allBlocks.length - 1].number,
+        }
+      : {
+        hash: "",
+        number: 0
+      };
+    return latestMinedBlock;
+  }
+
   // Get all available blocks on the same chain (sorted by block number)
-  public async getRecentBlocks(): Promise<BlockResponse[]> {
+  public async getRecentBlocks(blockIndex: number): Promise<BlockResponse[]> {
     const allBlocks = await this.blockStorage.getAllBlocks();
     const lastCommittedBlockInfo = await this.getLastSubnetCommittedBlock();
     const lastCommittedBlock = await this.blockStorage.getMinedBlockByHash(lastCommittedBlockInfo.hash);
+    const endIndex = blockIndex != -1? allBlocks.findIndex((block) => block.number == blockIndex) : allBlocks.length-1;
```

Comment:
My suggestion is use block num as the index instead of the index in the list.

Reply:
do you mean like
```
const selectedBlocks = allblocks.map((block), block.number < blockIndex && block.number >= blockIndex-50 )
```

Reply:
let me explain a bit in more details:
I think there is no need to keep those historical blocks in the memory, no one going to look at them and it will just add more complexity to the code if we keep them around.
So my suggestion is, show the first 20 in UI, but keep 50-100 blocks in our backend memory. If the UI is asking for blocks that is older than the 100th block we have in memory, then we just make proxy request to the nodes to get block num from 100 -> whatever user want. BUT, we don't keep them in the memory. It's useless for us.

Reply:
If you want to do this ` first 20 in UI, but keep 50 blocks in our backend memory`, you still need this piece of code. With different constant.
```
export const MAX_NUM_OF_BLOCKS_IN_HISTORY = 50;
export const NUM_OF_BLOCKS_RETURN = 20;
```

And I am asking fisher whether we allow user to keep receiving block till block 0 or not.